draft-ietf-lamps-rfc4210bis.md

```diff
@@ -1068,50 +1068,60 @@ A certification request message for a KEM certificate SHALL use POPOPrivKey by u
-end entities.  Self-signed CAs SHALL be considered as directly
-trusted CAs.  Recognizing whether a non-self-signed CA is supposed to
-be directly trusted for some end entities is a matter of CA policy
-and is thus beyond the scope of this document.
+end entities.  Recognizing whether a self-signed or non-self-signed
+CA is supposed to be directly trusted for some end entities is a
+matter of CA policy and end entity configuration. This is thus beyond
+the scope of this document.
 
 The basis of the procedure described here is that the CA protects its
-new public key using its previous private key and vice versa.  Thus,
-when a CA updates its key pair it must generate two extra
-cACertificate attribute values if certificates are made available
-using an X.500 directory (for a total of four: OldWithOld,
-OldWithNew, NewWithOld, and NewWithNew).
+new public key using its previous private key and vice versa. Thus,
+when a CA updates its key pair it may generate two link certificates
+"old with new" and "new with old".
+
+Note: The usage of link certificates has been shown to be very use
+case specific and no assumptions are done on this aspect.
+RootCaKeyUpdateContent is updated to specify these link certificates
+as optional.
+
+Note: When an LDAP directory is used to publish root CA updates, the
+old and new root CA certificates together with the two link
+certificates are stored as cACertificate attribute values.
 
 When a CA changes its key pair, those entities who have acquired the
-old CA public key via "out-of-band" means are most affected.  It is
-these end entities who will need access to the new CA public key
-protected with the old CA private key.  However, they will only
-require this for a limited period (until they have acquired the new
-CA public key via the "out-of-band" mechanism).  This will typically
-be easily achieved when these end entities' certificates expire.
+old CA public key via "out-of-band" means are most affected.  These
+end entities need to acquire the new CA public key in a trusted way.
+This may be achieved "out-of-band", by using a repository, or by
+using online messages also containing the link certificates
+"new with old". Once the end entity acquired and properly verified
+the new CA public key, it must load the new trust anchor information
+into its trusted store.
 
 The data structure used to protect the new and old CA public keys is
-a standard certificate (which may also contain extensions).  There
-are no new data structures required.
+typically a standard X.509 v3 self-signed certificate (which may also
+contain extensions).  There are no new data structures required.
 
-Note 1:  This scheme does not make use of any of the X.509 v3
-extensions as it must be able to work even for version 1
-certificates.  The presence of the KeyIdentifier extension would make
-for efficiency improvements.
+Note:  Sometime root CA certificates do not make use of any of the
```

Comment:
Some Language issues:

Original:
Note:  Sometime root CA certificates do not make use of any of the
X.509 v3 extensions and may be X.509 v1 certificates. Therefore, root
CA key update must be able to even for version 1 certificates.
Anyhow, to make use of the KeyIdentifier extension for easier path
building, the use of X.509 v3 is recommended.

Suggested:
Note:  Sometimes root CA certificates do not make use of 
X.509 v3 extensions and may be X.509 v1 certificates. Therefore, a root
CA key update must be able to work for version 1 certificates.
The use of the X.509 v3 KeyIdentifier extension is recommended for 
easier path building.

Reply:
I like the change, thanks.